astroquery/esa/hubble/core.py

```diff
@@ -179,60 +217,202 @@ def get_postcard(self, observation_id, calibration_level="RAW",
+            data. RAW (1), CALIBRATED (2), PRODUCT (3) or AUXILIARY (0)
+        data_product_type : str, optional
+            High level description of the product.
+            image, spectrum or timeseries.
+        intent : str, optional
+            The intent of the original observer in acquiring this observation.
+            SCIENCE or CALIBRATION
+        collection : list of str, optional
+            List of collections that are available in eHST catalogue.
+            HLA, HST
+        instrument_name : list of str, optional
+            Name(s) of the instrument(s) used to generate the dataset
+        filters : list of str, optional
+            Name(s) of the filter(s) used to generate the dataset
+        async_job : bool, optional, default 'False'
+            executes the query (job) in asynchronous/synchronous mode (default
+            synchronous)
+        filename : str, default None
+            Path and name of the file to store the results.
+            If the filename is defined, the file will be
+            automatically saved
+        output_format : string
+            results format. Options are:
+            'votable': str, binary VOTable format
+            'csv': str, comma-separated values format
+        save : bool
+            optional, default 'False'
+            Flag to save the result in a file. If the filename
+            is not defined, it will use a formatted name to save
+            the file
+        cache : bool
+            optional, default 'True'
+            Flag to save the results in the local cache
+        verbose : bool, optional, default 'False'
+            flag to display information about the process
+
+        Returns
+        -------
+        astropy.table.Table with the result of the cone_search
+        """
+        crit_query = self.query_criteria(calibration_level=calibration_level,
+                                         data_product_type=data_product_type,
+                                         intent=intent,
+                                         obs_collection=obs_collection,
+                                         instrument_name=instrument_name,
+                                         filters=filters,
+                                         async_job=True,
+                                         get_query=True)
+        if crit_query.endswith(")"):
+            crit_query = crit_query[:-1] + " AND "
         else:
-            fileobj = BytesIO(response.content)
-            table = Table.read(fileobj, format=output_format)
-            # TODO: add "correct units" material here
+            crit_query = crit_query + " WHERE ("
+
+        if target and coordinates:
+            raise TypeError("Please use only target or coordinates as"
+                            "parameter.")
+        if target:
+            try:
+                ra, dec = self._query_tap_target(target)
+            except Exception:
```

Comment:
does `_query_tap_target` raise a particular exception type?  Please catch only specific exception types; the generic `Exception` here will catch even `ctrl-c` from the user, which we shouldn't do.

Reply:
I have modified _query_tap_target to detect a key error when no result is available and raise a ValueError informing that the target cannot be resolved.